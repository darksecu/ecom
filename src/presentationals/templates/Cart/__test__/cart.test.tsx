import * as React from 'react';
import {View} from 'react-native';
import {CartTemplate} from '..';
import renderer from 'react-test-renderer';
import {fireEvent, render} from 'react-native-testing-library';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch,
  useState: () => mockDispatch,
}));

describe('PDP Template component', () => {
  it('SnapShot', () => {
    const expected = renderer.create(<CartTemplate />).toJSON();
    expect(expected).toMatchSnapshot();
  });

  it('Check is flatlist exist', async () => {
    const {findByTestId} = render(<CartTemplate />);
    const expected = await findByTestId('CartList');
    expect(expected).toBeDefined();
  });

  //   it('Quantity is 0 by default', async () => {
  //     const {findByTestId} = render(<PDPTemplate {...dummyData} />);
  //     const Quantity = await findByTestId('quanityDisplay');
  //     expect(Quantity.children).toEqual(['0']);
  //   });

  //   it('Price Label Validation', async () => {
  //     const {findByTestId} = render(<PDPTemplate {...dummyData} />);
  //     const data = await findByTestId('PDPPrice');
  //     expect(data.children).toEqual([`USD `, String(dummyData.price)]);
  //   });

  //   it('Name Label Validation', async () => {
  //     const {findByTestId} = render(<PDPTemplate {...dummyData} />);
  //     const data = await findByTestId('PDPName');
  //     expect(data.children).toEqual([`${dummyData.name}`]);
  //   });
});
