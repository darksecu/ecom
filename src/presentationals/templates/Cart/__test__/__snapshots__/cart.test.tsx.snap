// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PDP Template component SnapShot 1`] = `
<RCTScrollView
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
  testID="CartList"
  viewabilityConfigCallbackPairs={[]}
>
  <View />
</RCTScrollView>
`;
